### ISSUE TITLE
Assertion failed: (!tupleTy.containsPackExpansionType() && "can't extract elements from tuples containing pack expansions " "right now"), function extractElements, file RValue.cpp, line 705.

### ISSUE THREAD

**Issue by dnadoba** [apple/swift#67916]
```swift
protocol BarProtocol {}

struct Foo<each Bar: BarProtocol> {
    var bar: (repeat each Bar)
}

@resultBuilder
struct FooBuilder {
    public static func buildBlock<each Bar: BarProtocol>(
        _ components: (repeat each Bar)
    ) -> Foo<repeat each Bar> {
        Foo(bar: components)
    }
}
```

```
Assertion failed: (!tupleTy.containsPackExpansionType() && "can't extract elements from tuples containing pack expansions " "right now"), function extractElements, file RValue.cpp, line 705.
Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the crash backtrace.
Stack dump:
0.	Program arguments: /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2023-08-12-a.xctoolchain/usr/bin/swift-frontend -frontend -c -primary-file /Users/davidnadoba/Repositories/PlaygroundExecutbale/Sources/PlaygroundExecutbale/Executable.swift -emit-dependencies-path /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/Objects-normal/arm64/Executable.d -emit-const-values-path /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/Objects-normal/arm64/Executable.swiftconstvalues -emit-reference-dependencies-path /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/Objects-normal/arm64/Executable.swiftdeps -serialize-diagnostics-path /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/Objects-normal/arm64/Executable.dia -target arm64-apple-macos14.0 -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk "/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk" -I /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug -I "/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib" -F /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug/PackageFrameworks -F /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug/PackageFrameworks -F /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug/PackageFrameworks -F /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug/PackageFrameworks -F /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug/PackageFrameworks -F /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug/PackageFrameworks -F /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug/PackageFrameworks -F /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug/PackageFrameworks -F /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug -F "/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks" -no-color-diagnostics -enable-testing -g -module-cache-path /Users/davidnadoba/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 5 -enforce-exclusivity=checked -Onone -D SWIFT_PACKAGE -D DEBUG -D Xcode -warn-concurrency -external-plugin-path "/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/host/plugins#/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-plugin-server" -external-plugin-path "/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/local/lib/swift/host/plugins#/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-plugin-server" -serialize-debugging-options -package-name playgroundexecutbale -const-gather-protocols-file /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/Objects-normal/arm64/PlaygroundExecutbale_const_extract_protocols.json -empty-abi-descriptor -validate-clang-modules-once -clang-build-session-file /Users/davidnadoba/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -Xcc -working-directory -Xcc /Users/davidnadoba/Repositories/PlaygroundExecutbale -resource-dir /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2023-08-12-a.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -fmodule-map-file=/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/GeneratedModuleMaps/CNIOLinux.modulemap -Xcc -fmodule-map-file=/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/GeneratedModuleMaps/CNIOAtomics.modulemap -Xcc -I/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/SourcePackages/checkouts/swift-atomics/Sources/_AtomicsShims/include -Xcc -I/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/SourcePackages/checkouts/swift-nio/Sources/CNIOWindows/include -Xcc -I/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/SourcePackages/checkouts/swift-nio/Sources/CNIOLinux/include -Xcc -I/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/SourcePackages/checkouts/swift-nio/Sources/CNIOAtomics/include -Xcc -I/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Products/Debug/include -Xcc -I/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/DerivedSources-normal/arm64 -Xcc -I/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/DerivedSources/arm64 -Xcc -I/Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/DerivedSources -Xcc -DSWIFT_PACKAGE -Xcc -DDEBUG=1 -module-name PlaygroundExecutbale -frontend-parseable-output -disable-clang-spi -target-sdk-version 14.0 -target-sdk-name macosx14.0 -external-plugin-path "/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/lib/swift/host/plugins#/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/bin/swift-plugin-server" -external-plugin-path "/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/local/lib/swift/host/plugins#/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/bin/swift-plugin-server" -external-plugin-path "/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib/swift/host/plugins#/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/bin/swift-plugin-server" -external-plugin-path "/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/local/lib/swift/host/plugins#/Applications/Xcode 15.0 Beta6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/bin/swift-plugin-server" -plugin-path /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2023-08-12-a.xctoolchain/usr/lib/swift/host/plugins -plugin-path /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2023-08-12-a.xctoolchain/usr/local/lib/swift/host/plugins -parse-as-library -o /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Build/Intermediates.noindex/PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/Objects-normal/arm64/Executable.o -index-unit-output-path /PlaygroundExecutbale.build/Debug/PlaygroundExecutbale.build/Objects-normal/arm64/Executable.o -index-store-path /Users/davidnadoba/Library/Developer/Xcode/DerivedData/PlaygroundExecutbale-dnjdfgihtwpodwcvmuotcusgvyuw/Index.noindex/DataStore -index-system-modules
1.	Apple Swift version 5.9-dev (LLVM fd38736063c15cd, Swift a533c63d783f5b8)
2.	Compiling with the current language version
3.	While evaluating request ASTLoweringRequest(Lowering AST to SIL for file "/Users/davidnadoba/Repositories/PlaygroundExecutbale/Sources/PlaygroundExecutbale/Executable.swift")
4.	While silgen emitFunction SIL function "@$s20PlaygroundExecutbale10FooBuilderV10buildBlockyAA0C0VyxxQp_QPGxxQp_t_tRvzAA11BarProtocolRzlFZ".
 for 'buildBlock(_:)' (at /Users/davidnadoba/Repositories/PlaygroundExecutbale/Sources/PlaygroundExecutbale/Executable.swift:19:12)
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x0000000107348d80 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  swift-frontend           0x0000000107348124 llvm::sys::RunSignalHandlers() + 112
2  swift-frontend           0x00000001073493c0 SignalHandler(int) + 304
3  libsystem_platform.dylib 0x0000000185845a24 _sigtramp + 56
4  libsystem_pthread.dylib  0x0000000185816cc0 pthread_kill + 288
5  libsystem_c.dylib        0x0000000185726a80 abort + 180
6  libsystem_c.dylib        0x0000000185725d9c err + 0
7  swift-frontend           0x0000000107425ccc swift::Lowering::RValue::extractElements(llvm::SmallVectorImpl<swift::Lowering::RValue>&) && (.cold.3) + 0
8  swift-frontend           0x0000000102a99374 swift::Lowering::RValue::extractElements(llvm::SmallVectorImpl<swift::Lowering::RValue>&) && + 436
9  swift-frontend           0x0000000102abe834 (anonymous namespace)::ArgEmitter::emitExpanded(swift::Lowering::ArgumentSource&&, swift::Lowering::AbstractionPattern) + 280
10 swift-frontend           0x0000000102abeb4c (anonymous namespace)::ArgEmitter::emitExpanded(swift::Lowering::ArgumentSource&&, swift::Lowering::AbstractionPattern) + 1072
11 swift-frontend           0x0000000102abd070 (anonymous namespace)::ArgEmitter::emit(swift::Lowering::ArgumentSource&&, swift::Lowering::AbstractionPattern, llvm::Optional<swift::AnyFunctionType::Param>) + 164
12 swift-frontend           0x0000000102aafa90 (anonymous namespace)::ArgEmitter::emitSingleArg(swift::Lowering::ArgumentSource&&, swift::Lowering::AbstractionPattern, llvm::Optional<swift::AnyFunctionType::Param>) + 424
13 swift-frontend           0x0000000102abc91c (anonymous namespace)::ArgEmitter::emitPreparedArgs(swift::Lowering::PreparedArguments&&, swift::Lowering::AbstractionPattern) + 192
14 swift-frontend           0x0000000102ac6e28 (anonymous namespace)::CallSite::emit(swift::Lowering::SILGenFunction&, swift::Lowering::AbstractionPattern, swift::CanTypeWrapper<swift::SILFunctionType>, (anonymous namespace)::ParamLowering&, llvm::SmallVectorImpl<swift::Lowering::ManagedValue>&, llvm::SmallVectorImpl<(anonymous namespace)::DelayedArgument>&, swift::ForeignInfo const&) && + 536
15 swift-frontend           0x0000000102ac67a8 (anonymous namespace)::CallEmission::emitArgumentsForNormalApply(swift::Lowering::AbstractionPattern, swift::CanTypeWrapper<swift::SILFunctionType>, swift::ForeignInfo const&, llvm::SmallVectorImpl<swift::Lowering::ManagedValue>&, llvm::Optional<swift::SILLocation>&) + 1112
16 swift-frontend           0x0000000102ab3230 (anonymous namespace)::CallEmission::apply(swift::Lowering::SGFContext) + 2288
17 swift-frontend           0x0000000102ab1660 swift::Lowering::SILGenFunction::emitApplyExpr(swift::ApplyExpr*, swift::Lowering::SGFContext) + 1996
18 swift-frontend           0x0000000102affa88 swift::Lowering::SILGenFunction::emitExprInto(swift::Expr*, swift::Lowering::Initialization*, llvm::Optional<swift::SILLocation>) + 248
19 swift-frontend           0x0000000102b79540 swift::Lowering::SILGenFunction::emitReturnExpr(swift::SILLocation, swift::Expr*) + 796
20 swift-frontend           0x0000000102b7776c swift::ASTVisitor<(anonymous namespace)::StmtEmitter, void, void, void, void, void, void>::visit(swift::Stmt*) + 8560
21 swift-frontend           0x0000000102b7669c swift::ASTVisitor<(anonymous namespace)::StmtEmitter, void, void, void, void, void, void>::visit(swift::Stmt*) + 4256
22 swift-frontend           0x0000000102b755f0 swift::Lowering::SILGenFunction::emitStmt(swift::Stmt*) + 24
23 swift-frontend           0x0000000102b214dc swift::Lowering::SILGenFunction::emitFunction(swift::FuncDecl*) + 408
24 swift-frontend           0x0000000102aa36f4 swift::Lowering::SILGenModule::emitFunctionDefinition(swift::SILDeclRef, swift::SILFunction*) + 9180
25 swift-frontend           0x0000000102aa3f88 swift::Lowering::SILGenModule::emitOrDelayFunction(swift::SILDeclRef) + 204
26 swift-frontend           0x0000000102aa1300 swift::Lowering::SILGenModule::emitFunction(swift::FuncDecl*) + 164
27 swift-frontend           0x0000000102b8553c (anonymous namespace)::SILGenType::visitFuncDecl(swift::FuncDecl*) + 28
28 swift-frontend           0x0000000102b82484 (anonymous namespace)::SILGenType::emitType() + 252
29 swift-frontend           0x0000000102b8237c swift::Lowering::SILGenModule::visitNominalTypeDecl(swift::NominalTypeDecl*) + 24
30 swift-frontend           0x0000000102aa6738 swift::ASTLoweringRequest::evaluate(swift::Evaluator&, swift::ASTLoweringDescriptor) const + 976
31 swift-frontend           0x0000000102b7508c swift::SimpleRequest<swift::ASTLoweringRequest, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>> (swift::ASTLoweringDescriptor), (swift::RequestFlags)9>::evaluateRequest(swift::ASTLoweringRequest const&, swift::Evaluator&) + 148
32 swift-frontend           0x0000000102aa953c llvm::Expected<swift::ASTLoweringRequest::OutputType> swift::Evaluator::getResultUncached<swift::ASTLoweringRequest>(swift::ASTLoweringRequest const&) + 396
33 swift-frontend           0x0000000102aa6f20 swift::performASTLowering(swift::FileUnit&, swift::Lowering::TypeConverter&, swift::SILOptions const&, swift::IRGenOptions const*) + 100
34 swift-frontend           0x000000010253c9fc swift::performCompileStepsPostSema(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 556
35 swift-frontend           0x000000010254bdd4 withSemanticAnalysis(swift::CompilerInstance&, swift::FrontendObserver*, llvm::function_ref<bool (swift::CompilerInstance&)>, bool) + 160
36 swift-frontend           0x000000010253f584 performCompile(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 740
37 swift-frontend           0x000000010253e508 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 2496
38 swift-frontend           0x0000000102377f04 swift::mainEntry(int, char const**) + 2144
39 dyld                     0x000000018549d058 start + 2224
```

with latest nightly toolchain `swift-DEVELOPMENT-SNAPSHOT-2023-08-12-a`

**Comment by dnadoba** [comment]
rdar://113862745

**Comment by vanvoorden** [comment]
https://gist.github.com/vanvoorden/91038b4d0a6ebf31c55ae21e88158c84

@dnadoba I saw something similar here… but at least I also have this workaround that seems to unblock me for now.

**Comment by AnthonyLatsis** [comment]
Duplicate of #66392
